packages/suite/src/utils/suite/logsUtils.ts

```diff
@@ -102,34 +128,63 @@ export const redactAction = (action: Action): Action => {
     };
 };
 
-export const redactCustom = (action: CustomLogEntry['action']) => {
-    let payload;
-    switch (action.type) {
-        case SUITE.AUTH_DEVICE:
-            payload = {
-                state: REDACTED_REPLACEMENT,
-                payload: redactDevice(action.payload),
-            };
-            break;
-        case TRANSACTION.ADD:
-            payload = {
-                ...action.payload,
-                deviceState: REDACTED_REPLACEMENT,
-                descriptor: REDACTED_REPLACEMENT,
-            };
-            break;
-        case ACCOUNT.UPDATE_SELECTED_ACCOUNT:
-            payload = {
-                ...action.payload,
-                account: redactAccount(action.payload?.account),
-            };
-            break;
+export const getApplicationLog = (log: LogEntry[], redactSensitiveData = false) =>
+    log.map(entry => {
+        const metadata = {
+            type: entry.type,
+            datetime: entry.datetime,
+        };
 
-        default:
-            return action;
-    }
-    return {
-        ...action,
-        payload,
-    };
-};
+        let redactedAction = entry.payload;
+        if (redactSensitiveData) {
+            redactedAction = redactAction(entry);
+        }
+
+        if (typeof redactedAction?.payload === 'object') {
```

Comment:
Are you checking for payload being there or payload being an object? If the first one then why not simply 

```
       return {
            ...metadata,
            payload: {
                ...(redactedAction?.payload || redactedAction),
            },
        };
```

Reply:
I am checking if payload is an object because some actions have a payload of type `number` or `string`

Reply:
It's just in one case you are spreading `redactedAction.payload` into payload and in the other - the whole `redactedAction`, so I got confused